src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -88,15 +87,62 @@
 
 @postgres_dialect.segment(replace=True)
 class FunctionDefinitionGrammar(BaseSegment):
-    """This is the body of a `CREATE FUNCTION AS` statement."""
+    """This is the body of a `CREATE FUNCTION AS` statement.
+
+    Options supported as defined in https://www.postgresql.org/docs/9.1/sql-createfunction.html
+    """
 
     match_grammar = Sequence(
-        "AS",
-        OneOf(Ref("QuotedLiteralSegment"), Ref("DollarQuotedLiteralSegment")),
+        AnyNumberOf(
+            Sequence("TRANSFORM", "FOR", "TYPE", Ref("ParameterNameSegment")),
+            OneOf("WINDOW", "IMMUTABLE", "STABLE", "VOLATILE", "STRICT"),
+            Sequence(OneOf("NOT", optional=True), "LEAKPROOF"),
+            Sequence(
+                OneOf("CALLED", Sequence("RETURNS", "NULL")), "ON", "NULL", "INPUT"
+            ),
+            Sequence(
+                OneOf("EXTERNAL", optional=True),
+                "SECURITY",
+                OneOf("INVOKER", "DEFINER"),
```

Comment:
Again should we just use some repetition here to align more with the spec even if it is slightly more code? WDYT?

Reply:
I personally think it's neater this way, and not particularly confusing. Happy to defer to your opinion if you feel strongly about it.

Reply:
My concern is not necessarily understanding what you've written, but when the next version of Postgres comes out with one extra line of syntax - will it be obvious where and how to insert that new one? I think being closer to the spec will make that process easier as a future contributor can see exactly what's been implemented already and therefore what's missing. It took me a bit of matching your code to the spec to understand what you'd implemented was correct.

Reply:
I think I agree with you on the CALLED ON NULL INPUT set, but constructing the SECURITY makes more sense to me. I've implemented the rest of your changes, can you see if it makes more sense, now that the thing as a whole marries up with the Postgres spec?

Reply:
Yup that's fine as more obvious. Looks great now. Just one more comment on the test cases then think we're good to merge!